### ISSUE TITLE
VK_NUMPADx key events are not interpreted as numbers if numlock is disabled

### ISSUE THREAD

**Issue by Rojetto** [microsoft/microsoft-ui-xaml#5008]
**Describe the bug**
I maintain a [driver](https://github.com/Rojetto/ReNeo/issues/32) for the [Neo ergonomic keyboard layout](https://neo-layout.org/) (documentation is unfortunately all in German). Due to the complex layering system we need to remap all keys (including the Numpad) using a low level keyboard hook. Because the Windows keyboard stack sends weird fake Shift key events when Numlock is enabled, we automatically disable it, as we have our own mechanism for switching between numbers and navigation keys anyway. With Numlock disabled however, WinUI apps (Calculator, Mail, Photos, ...) interpret these events as cursor movements instead of numbers. All other applications I've encountered (including older Win32 programs) don't behave this way and instead seemingly rely on the native keyboard stack to handle numlock state.

**Steps to reproduce the bug**
Disable Numlock and somehow send a `VK_NUMPADx` key event to a WinUI app. No number appears, but the text cursor moves.

**Expected behavior**
`VK_NUMPADx` events result in numbers. Only `VK_LEFT`/`VK_RIGHT`/`VK_UP`/`VK_DOWN` events with scancode and extended bit set appropriately should be interpreted as numpad navkeys.

**Version Info**
I don't know what WinUI version each of these apps ship with.

**Additional context**
I first encountered and described this issue in https://github.com/microsoft/calculator/issues/1537.

**Comment by StephenLPeters** [comment]
@Austin-Lamb FYI
